process/track/sovereignChainBlockProcessor.go

```diff
@@ -86,9 +88,32 @@ func (scbp *sovereignChainBlockProcessor) doJobOnReceivedCrossNotarizedHeader(sh
 }
 
 func (scbp *sovereignChainBlockProcessor) requestHeaderWithShardAndNonce(shardID uint32, nonce uint64) {
-	if shardID == core.SovereignChainShardId {
+	if shardID == scbp.shardCoordinator.SelfId() {
+		scbp.requestHandler.RequestShardHeaderByNonce(shardID, nonce)
+	} else if shardID == core.MainChainShardId {
 		scbp.extendedShardHeaderRequester.RequestExtendedShardHeaderByNonce(nonce)
 	} else {
-		scbp.requestHandler.RequestShardHeaderByNonce(shardID, nonce)
+		log.Warn("sovereignChainBlockProcessor.requestHeaderWithShardAndNonce requested header for unknown shard",
+			"shardID", shardID, "nonce", nonce)
+	}
+}
+
+func (scbp *sovereignChainBlockProcessor) requestHeadersIfNothingNewIsReceived(
+	lastNotarizedHeaderNonce uint64,
+	latestValidHeader data.HeaderHandler,
+	highestRoundInReceivedHeaders uint64,
+	shardID uint32,
+) {
+	if check.IfNil(latestValidHeader) {
+		return
 	}
+	// double-checking here for both shardID and header type just to be safe
+	if shardID == core.MainChainShardId {
```

Comment:
I would go with chainID - not shardID.

Reply:
This func is called with shardID == either selfID() or MainChainShardID.
We do not use chanID